### ISSUE TITLE
Warn or throw error when signal.value is read untracked during component render

### ISSUE THREAD

**Issue by justjake** [tldraw/signia#75]
Currently users can write a function component that looks like this, that appears to work on first render or if a parent subscribes to an atom, but actually isn't reactive to atom changes. The example [comes from Discord](https://discord.com/channels/859816885297741824/1078634013553602590/1100468281103626332):

```tsx
const Prova = ({name}: {name: Atom<string>})=>{
  return <>
  <input type="text" value={name.value} onChange={(e)=>name.set(e.target.value)}/>
  <div>Your name is {name.value}</div>
  </>
}
```

Signia and/or Signia-react should either log a warning or throw an error if a component reads an atom during render in a non-reactive way. [Starbeam implements this](https://github.com/starbeamjs/starbeam/blob/97cfcfa6bec298c5774e962c9755f4bb6e723593/packages/react/use-strict-lifecycle/src/react.ts#L69-L83) by checking `React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner.current`, which is only set when in development versions of React.

In Notion's internal state management system, we throw an error when this happens.

**Comment by ds300** [comment]
Yeah this feels like it could save a lot of frustration! I'm down for adding a `console.warn`, maybe with an option to turn it into a thrown error if folks wanna get strict.
